### PR TITLE
Fixes 4825: add template attribute for when env is created

### PR DESCRIPTION
**Pull request by jlsherrill**
## Summary

adds an rhsm_environment_created db column and api attribute.  It is false when a template is first created.  Once it is created via the update-template-content task, it gets set to true.

## Testing steps

1.  create and snapshot one repository
2.  Create a template with the repository:
```
####
POST http://localhost:8000/api/content-sources/v1.0/templates/
x-rh-identity: eyJpZGVudGl0eSI6eyJvcmdfaWQiOiI5IiwgInR5cGUiOiJVc2VyIiwidXNlciI6eyJ1c2VybmFtZSI6ImZvbyJ9LCJhY2NvdW50X251bWJlciI6ImZvbyIsImludGVybmFsIjp7Im9yZ19pZCI6IjkifX19Cg==
Content-Type: application/json

{
  "name":"test7",
  "arch": "x86_64",
  "version": "9",
  "repository_uuids": ["270fabbd-ca28-4cd0-9cff-a274b4e797b9"],
  "date": "2024-09-10T15:09:43-04:00"
}
```

3.  check teh return:

  "rhsm_environment_created": false

4.  fetch the created template:
```
###
GET http://localhost:8000/api/content-sources/v1.0/templates/687c93e2-b146-428d-b65d-a972bed5edbe
x-rh-identity: eyJpZGVudGl0eSI6eyJvcmdfaWQiOiI5IiwgInR5cGUiOiJVc2VyIiwidXNlciI6eyJ1c2VybmFtZSI6ImZvbyJ9LCJhY2NvdW50X251bWJlciI6ImZvbyIsImludGVybmFsIjp7Im9yZ19pZCI6IjkifX19Cg==
//x-Rh-Insights-Request-Id: 9d229d34-7219-405d-ba3d-8f99cf7c36ae
Content-Type: application/json
```

after a few seconds of creation, it should now be true:
```
  "rhsm_environment_created": true
```

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-4825

File: db/migrations/20241011084507_AddEnvironmentCreated.up.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+BEGIN;
+
+alter table templates add column rhsm_environment_created boolean NOT NULL DEFAULT false;
+update templates set rhsm_environment_created = true;
```
Comment (xbhouse) [inline db/migrations/20241011084507_AddEnvironmentCreated.up.sql:4]: existing templates *should* have had an environment created, but in the odd case they haven't, do you think we should check that at least one update-template-content task has completed before setting this to true?

**Comment by xbhouse** [comment]
just the one question, overall looks good and works well!

**Comment by mayurilahane** [comment]
Created a repository 

```
In [1]: app.content_sources.rest_client.repositories_api.create_repository(dict(name='test', origin='upload', snapshot=True))

Out[1]: 
{'account_id': '0369233',
 'content_type': 'rpm',
 'distribution_arch': 'any',
 'distribution_versions': ['any'],
 'failed_introspections_count': 0,
 'gpg_key': '',
 'label': 'test',
 'last_introspection_error': '',
 'last_introspection_status': 'Valid',
 'last_introspection_time': '2024-10-17T09:16:08Z',
 'last_snapshot_task_uuid': '9263a30f-142a-4694-9316-9cb37b4f5209',
 'last_success_introspection_time': '2024-10-17T09:16:08Z',
 'last_update_introspection_time': '2024-10-17T09:16:08Z',
 'metadata_verification': False,
 'module_hotfixes': False,
 'name': 'test',
 'org_id': '3340851',
 'origin': 'upload',
 'package_count': 0,
 'snapshot': True,
 'status': 'Valid',
 'url': '',
 'uuid': 'f5b987cc-aecb-4373-b6ad-f38b5c4f2cac'}
```

File: db/migrations/20241011084507_AddEnvironmentCreated.up.sql (hunk 1 above)
Comment (jlsherrill) [reply db/migrations/20241011084507_AddEnvironmentCreated.up.sql:4]: I think this should be pretty easy to do in a query, let me see

Comment (jlsherrill) [reply db/migrations/20241011084507_AddEnvironmentCreated.up.sql]: updated query here:

https://github.com/content-services/content-sources-backend/pull/846/files#diff-cbb9bfb9e61fbfb4cc6348f60e645c0f16ff1d42c43ce7621eb4c3dd5cc2f7deR4-R5

**Review by xbhouse — APPROVED** [review]

**Comment by mayurilahane** [comment]
lgtm!

**Comment by mayurilahane** [comment]
Verified again on stage after merging this PR. 
The get_template function now returns 
`rhsm_environment_created: True` after some time

**Pull request merged**